packages/react-data-grid/src/Grid.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { isValidElementType } from 'react-is';
+
+import Header from './Header';
+import Viewport, { ViewportProps, ScrollState } from './Viewport';
+import { isFrozen } from './ColumnUtils';
+import { Column } from './common/types';
+import { SCROLL_DIRECTION, DEFINE_SORT } from './common/enums';
+
+interface Props extends ViewportProps {
+  columns: Column[];
+  headerRows: unknown[] | Function;
+  emptyRowsView?: React.ComponentType;
+  sortColumn?: string;
+  sortDirection: SCROLL_DIRECTION;
+  rowOffsetHeight: number;
+  draggableHeaderCell?: React.ComponentType;
+  scrollLeft?: number;
+  onViewportKeydown(e: React.KeyboardEvent<HTMLDivElement>): void;
+  onViewportKeyup(e: React.KeyboardEvent<HTMLDivElement>): void;
+  onColumnResize(index: number, width: number): void;
+  onSort(columnKey: string, direction: DEFINE_SORT): void;
+  getValidFilterValues?(): unknown;
+  onHeaderDrop?(): void;
+}
+
+export default class Grid extends React.Component<Props> {
```

Comment:
Looks like it's safe and easy to turn this one into a functional component.
We can replace the refs with `useRef`, and `_scrollLeft` with... `useRef`.
https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

Reply:
Let's revisit this after we fix the scroll logic